File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    let map = {}
    for(let i = 1; i <= elements.length; i++){
        let size = 0;
        for(let k = 0; k < i; k++){
            size += elements[k];
        }
    
        map[size] = (map[size] ?? 0) + 1;
        
        for(let k = 0; k < elements.length; k++){
            size -= elements.at(k)
            size += elements.at((k + i) % elements.length)
            map[size] = (map[size] ?? 0) + 1;
        }
    }
    return Object.keys(map).length;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    dic = {}
    for i in range(len(elements)):
        size = sum(elements[:i])
        dic[size] = 0
        for k in range(len(elements)):
            size -= elements[k]
            size += elements[(k + i) % len(elements)]
            dic[size] = 0
    return len(dic.keys())